### PR TITLE
Add documentatio for PR 156

### PR DESCRIPTION
**Pull request by a4z**
Add docs for `@deprecated` commends, introduced in #156

### CODE REVIEW

**Comment by a4z** [comment]
If possible, @eakoli please have a look 
(I invited you into the github organization so I can request a review from you)

File: docs/idl.md [hunk 1]
```diff
@@ -327,3 +327,43 @@ will be `RecordWithConst::CONST_VALUE` in C++, `RecordWithConst.CONST_VALUE` in
 If comments are placed on top or inside a type definition, they will be converted to
 Javadoc / Doxygen compatible comments in the generated Java, C++, Objective-C and C++/CLI interfaces, or a
 Python docstring.
+
+## Deprecation Comments
+
+It's possible to add deprecation comments to IDL types and methods.
+
+The comment in the IDL file has to start the `# @deprecated reason/hing message` .
```
Comment (freitass) [inline docs/idl.md:335]: Out of curiosity, what's a "hing message"?

File: docs/idl.md [hunk 2]
```diff
@@ -327,3 +327,43 @@ will be `RecordWithConst::CONST_VALUE` in C++, `RecordWithConst.CONST_VALUE` in
...
+
+This will generate the corresponding deprecation annotation in the target language.
+
+```idl
+# @deprecated Use someother interface
```
Comment (freitass) [inline docs/idl.md:340]: If `someother` is meant to be a type we should follow Djinni's naming convention and write it as `some_other`.

**Review by freitass — APPROVED** [review]
Thanks for improving the docs!

**Review by jothepro — APPROVED** [review]
nice, thx! :)

File: docs/idl.md (hunk 2 above)
Comment (a4z) [reply docs/idl.md:340]: no, it's just something different, a placeholder for some error message, but maybe the fix is a good idea

File: docs/idl.md (hunk 1 above)
Comment (a4z) [reply docs/idl.md:335]: very probably a hint :-)

Comment (a4z) [reply docs/idl.md:335]:
```suggestion
The comment in the IDL file has to start the `# @deprecated reason/hint message` .
```

File: docs/idl.md (hunk 2 above)
Comment (a4z) [reply docs/idl.md:340]:
```suggestion
# @deprecated Use some_other interface
```

**Comment by a4z** [comment]
Thanks for the feedback! updated the unclear parts

**Pull request merged**